enhancements/microshift/advertise-version-details.md

```diff
@@ -0,0 +1,302 @@
+  version: ""
+```
+
+- `product` should hold a representation of the OpenShift variant
+  name. For example, `OCP` for OpenShift in standalone or external
+  control-plane configurations of all topologies and `MicroShift` for
+  MicroShift instances.
+- `major` should hold the major version number of the product as a
+  string. For example, for `4.12.0` the value would be `"4"`.
+- `minor` should hold the minor version number of the product as a
+  string. For example, for `4.12.0` the value would be `"12"`.
+- `version` should hold the complete version of the product as a
+  string. For example, for `4.12.0` the value would be `"4.12.0"`. The
+  version string may also include other qualifiers in nightly,
+  sprintly, CI, or hot-fix builds. For example,
+  `4.11.0-0.nightly-2022-06-23-153912`.
+
+The `version` string is meant for display in user interface
+situations.  The `major` and `minor` values are included separately so
+that API users do not have to parse the `version` string before
+applying rules based on the values.
+
+### Workflow Description
+
+#### Determining the Version of a Cluster via the API
+
+API clients can read the `openshift/version` ConfigMap contents in the
+normal way.
+
+#### Determining the Version of a Cluster via the CLI
+
+The `oc login` command reports the `message` value of the
+`openshift/motd` ConfigMap when a user successfully authenticates. The
+command should be extended to include the `version` string of the
+`openshift/version` ConfigMap, when it is present.
+
+The `oc version` command reports version details today. Some of the
+information it reports comes from the ClusterVersion resource, which
+is not visible to all users. The command should be updated to look at
+the `openshift/version` ConfigMap when it is present and the
+ClusterVersion resource cannot be read.
+
+#### Publishing the Version in MicroShift
+
+MicroShift will update the `openshift/version` ConfigMap contents on
+startup using a value compiled into the main binary. Upgrading
+MicroShift requires restarting the service, so there is no need to
+continue reconciling the contents of the ConfigMap.
+
+#### Publishing the Version in OCP
+
+OpenShift running in standalone or external control-plane
+configurations may be upgraded dynamically in place without loss of
+API connectivity. The cluster-version-operator will therefore update
+the contents of the ConfigMap as the version of the cluster
+changes. During an upgrade, we want to report the current version as
+the one from which the cluster is upgrading and not expose that an
+upgrade is in progress. The ConfigMap should therefore always contain
+the value of the most recent version marked as `Completed` in the
+history.
+
+### API Extensions
+
+None
+
+### Risks and Mitigations
+
+There is some risk that an API caller could use the version
+information to determine whether the cluster is vulnerable to a known
+exploit. API callers will need to be authenticated to read the data,
```

Comment:
How is this different than someone know the version of OCP API or Kube API being used in a cluster? I think if someone has access to the cluster they can retrieve enough information about the cluster anyway. So this does not change that.

Reply:
Right, I've included this because it came up in conversation elsewhere but I don't consider it a significant risk.